moshi/src/main/java/com/squareup/moshi/internal/Util.java

```diff
@@ -169,38 +171,52 @@ public static Type removeSubtypeWildcard(Type type) {
   }
 
   public static Type resolve(Type context, Class<?> contextRawType, Type toResolve) {
+    return resolve(context, contextRawType, toResolve, new HashSet<TypeVariable>());
```

Comment:
nit: https://publicobject.com/2016/02/08/linkedhashmap-is-always-better-than-hashmap/

Reply:
I don‘t love this solution because it allocates a collection always even though this case is quite rare.

Reply:
What if we initialize with an empty capacity? Still an instance but minimizes the footprint for non-typevar users

Reply:
lets do that, but use `null` as the sentinel for empty?